docs/en/observability/monitor-kubernetes.asciidoc

```diff
@@ -0,0 +1,223 @@
+software itself). Containers are a moving target, etc. These blog posts have
+some good introductory info:
+https://www.elastic.co/blog/kubernetes-observability-tutorial-k8s-log-monitoring-and-analysis-elastic-stack[here]
+and https://www.elastic.co/blog/monitoring-kubernetes-and-docker-containers-with-beats-logs-metrics-and-metadata[here].
+
+* Provide an overview of monitoring capabilities, perhaps with a diagram. Might
+want to show an example deployment that’s running a couple of applications, like
+Nginx and MySQL. Describe the data to collect and the technology stack for
+collecting the data. Probably best to keep this as a broad overview and explore
+the finer details later, but should touch on these concepts:
+
+** Deployment: What is a DaemonSet? Why deploy as a DaemonSet?
+
+** Filebeat for log collection
+
+** Metricbeat for metrics collection
+
+** Metadata for correlating logs, metrics, and traces:
+add_cloud_metada, add_docker_metada, add_kubernetes_metadata
+
+** Autodiscovery for automatically detecting and monitoring services as they appear
+dynamically. To avoid too much front loading, probably shouldn't talk about
+details like hints-based autodiscovery, labels, and templates, but will need to
+cover that some later.
+
+
+[discrete]
+== Part 1: Monitor logs
+
+TODO: Provide a brief intro. We'll need to decide how much to cover here vs
+the monitoring overview. Again lots of good introductory info here:
+https://www.elastic.co/blog/kubernetes-observability-tutorial-k8s-log-monitoring-and-analysis-elastic-stack)
+
+[discrete]
+=== Logs you should monitor
+
+TODO: Discuss the various logs (Kubernetes cluster, nodes, host, container logs)
+that users might want to collect. I'm not sure if this deserves a separate
+section or should be part of the previous section.
+
+[discrete]
+=== Deploy {filebeat} to collect logs
+
+TODO: Describe how to configure and deploy Filebeat as a daemonset. We should
+walk users through the parts of the deployment YAMLs, explain what they do, and point
+out the settings they might want to change or add. Rough list of what
+we need to cover:
+
+* Download the Filebeat deployment manifest
+
+* Set the connection information for Elasticsearch (also describe how to create
+secrets)
+
+* Configure log collection:
+
+** Collect Kubernetes logs (cluster, node)
+
+** Collect host logs
+
+** Collect container logs (use autodiscovery)
+We should probably explain what the default config with the container input does
+then focus on documenting how to use autodiscovery. 
+
+* Add metadata to events. Enrich the event with metadata coming from Docker,
+Kubernetes, host, and the cloud providers.
+
+* (optional) Drop unwanted events
+
+* Deploy Filebeat as a DaemonSet on Kubernetes
+
+[discrete]
+=== View logs in {kib}
+
+TODO: Describe how to view logs in Kibana. Show how to use the Logs app and how to
+view pre-built dashboards and visualizations.
```

Comment:
Note: I'm not sure if we enable Dashboard's loading  through the current manifest's but maybe we can add it.

Reply:
We could add a setup step to these docs, if necessary.